### ISSUE TITLE
Resource templating documentation has broken link to twisted.web.server.Resource which emphatically does not exist

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4968]
|[<img alt="lvh's avatar" src="https://avatars.githubusercontent.com/u/97816?s=50" width="50" height="50">](https://github.com/lvh)| @lvh reported|
|-|-|
|Trac ID|trac#4968|
|Type|defect|
|Created|2011-03-20 02:35:59Z|
|Branch|https://github.com/twisted/twisted/tree/broken-doc-link-4968|

[Resource templating](http://twistedmatrix.com/documents/current/web/howto/resource-templates.html) documentation points to API docs for `twisted.web.server.Resource` when it should obviously point to the `twisted.web.resource.Resource`. The latter has the benefit of actually existing, instead of the current thing which produces a broken link.

Spotted in [#4951](https://github.com/twisted/twisted/issues/4951).

<details><summary>Searchable metadata</summary>

```
trac-id__4968 4968
type__defect defect
reporter__lvh lvh
priority__normal normal
milestone__ 
branch__branches_broken_doc_link_4968 branches/broken-doc-link-4968
branch_author__tenth tenth
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation_easy documentation easy
time__1300588559000000 1300588559000000
changetime__1300666849000000 1300666849000000
version__None None
owner__tenth tenth
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tenth10th's avatar" src="https://avatars.githubusercontent.com/u/854235?s=50" width="50" height="50">](https://github.com/tenth10th)<a name="note_2"></a>|@tenth10th commented|
|-|-|

(In [31280]) Branching to 'broken-doc-link-4968'

**Comment by twisted-trac** [comment]
|[<img alt="tenth10th's avatar" src="https://avatars.githubusercontent.com/u/854235?s=50" width="50" height="50">](https://github.com/tenth10th)<a name="note_3"></a>|@tenth10th commented|
|-|-|

I have changed the API link from "twisted.web.server.Resource" to "twisted.web.resource.Resource".

While I was in there, I also fixed the broken link to the Quixote PTL page (which is now hosted on quixote.python.ca).

The Divmod Nevow link is also sort of broken, in the sense that the Divmod site is mostly unavailable, but there isn't really a solution for that currently, so I'm leaving it as-is for now.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @tenth10th|
|-|-|

Thanks for fixing this, tenth.

Your changes look great and ready to merge. If you're up for it:

 * On Debian, the package is now python-quixote
 * "that means using Python 2.2". Since we're so far away from 2.2 now, maybe just don't even mention the Python version restriction.
 * The references to ResourceScript and ResourceTemplate could use some markup or API linkification.

**Comment by twisted-trac** [comment]
|[<img alt="tenth10th's avatar" src="https://avatars.githubusercontent.com/u/854235?s=50" width="50" height="50">](https://github.com/tenth10th)<a name="note_6"></a>|@tenth10th removed owner|
|-|-|

Those changes all sound good, though I had to figure out a few things about PyDoctor and Lore. (But good things to know if I want to help out with docs.)

The Debian package is now python-quixote.

Removed the mention of Python 2.2

The first mentions of ResourceScript and ResourceTemplate are now API links, with the pluralization and phrasing fixed to reflect the singular names of the functions. (ResourceTemplate is currently undocumented, but at least it's linked for when the docs do get added...)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_8"></a>|@jesstess set owner to @tenth10th|
|-|-|

Sorry, I should have run a Sphinx build from the get-go. The [Sphinx doc build](http://buildbot.twistedmatrix.com/builds/sphinx-html/267-15836/projects/web/howto/resource-templates.html) reveals a few things:

 * The new API links to ResourceScript and ResourceTemplate show up as "api:twisted.web.script.ResourceScript <ResourceScript>". Adding a space before the code tag should fix this.
 * The link to "thequixote web site" needs a space before "quixote".

Other than that, this is good to merge :)

**Comment by twisted-trac** [comment]
|[<img alt="tenth10th's avatar" src="https://avatars.githubusercontent.com/u/854235?s=50" width="50" height="50">](https://github.com/tenth10th)<a name="note_9"></a>|@tenth10th commented|
|-|-|

Given how many Sphinx "missing space" tickets I reviewed at the last sprint, I really should have thought to run it myself!

But I've fixed the missing spaces and confirmed that [the changes look okay](http://buildbot.twistedmatrix.com/builds/sphinx-html/272-15837/projects/web/howto/resource-templates.html) in Sphinx as well.

**Comment by twisted-trac** [comment]
|[<img alt="tenth10th's avatar" src="https://avatars.githubusercontent.com/u/854235?s=50" width="50" height="50">](https://github.com/tenth10th)<a name="note_10"></a>|@tenth10th set status to `closed`|
|-|-|

(In [31312]) Merge broken-doc-link-4968: Fixed broken links and outdated info in resource-templates.xhtml

Author: tenth

Reviewer: jesstess

Fixes: [#4968](https://github.com/twisted/twisted/issues/4968)

Changed the API link from "twisted.web.server.Resource" to "twisted.web.resource.Resource"

Fixed a broken link to the Quixote PTL page

Changed the Debian quixote package name to "python-quixote"

Removed mention of Python 2.2 requirements (for now)

Made the first mentions of "ResourceScript" and "ResourceTemplate" into API links

(Fixed some missing spaces in the above changes and confirmed proper Sphinx rendering)
